README.md

```diff
@@ -560,6 +560,157 @@ In the following example, we have defined multiple imposters for the `POST /goph
 ]
 ````
 
+#### Using Templating in Responses
+Killgrave supports templating in responses, allowing you to create dynamic responses based on request data. This feature uses Go's `text/template` package to render templates.
+
+In the following example, we define an imposter for the `GET /gophers/{id}` endpoint. The response body uses templating to include the id from the request URL.
+
+````json
+[
+  {
+    "request": {
+        "method": "GET",
+        "endpoint": "/gophers/{id}"
+    },
+    "response": {
+        "status": 200,
+        "headers": {
+            "Content-Type": "application/json"
+        },
+        "body": "{\"id\": \"{{.PathParams.id}}\", \"name\": \"Gopher\"}"
+    }
+  }
+]
+````
+In this example:
+
+- The endpoint field uses a path parameter {id}.
+- The body field in the response uses a template to include the id from the request URL.
+
+You can also use other parts of the request in your templates, such query parameters and the request body.
+Since query parameters can be used more than once, they are stored in an array and you can access them by index or use the `stringsJoin` function to concatenate them.
+
+Here is an example that includes query parameters gopherColor and gopherAge in the response, one of which can be used more than once:
+
+````jsonc
+// expects a request to, for example, GET /gophers/bca49e8a-82dd-4c5d-b886-13a6ceb3744b?gopherColor=Blue&gopherColor=Purple&gopherAge=42
+[
+  {
+    "request": {
+        "method": "GET",
+        "endpoint": "/gophers/{id}",
+        "params": {
+            "gopherColor": "{v:[a-z]+}",
+            "gopherAge": "{v:[0-9]+}"
+        }
+    },
+    "response": {
+        "status": 200,
+        "headers": {
+            "Content-Type": "application/json"
+        },
+        "body": "{\"id\": \"{{ .PathParams.id }}\", \"color\": \"{{ stringsJoin .QueryParams.gopherColor "," }}\", \"age\": {{ index .QueryParams.gopherAge 0 }}}"
```

Comment:
Out of curiosity; why do we need `index.QueryParams` and it cannot be just `.QueryParams`, as with `.PathParams`? 🤔

Reply:
Is that just a typo? I see for `gopherColor` you just used `.QueryParams`.

Reply:
The thing is, query parameters are `map[string][]string` meaning each key can have multiple items by querying like this `?gopherColor=Blue&gopherColor=Purple`.

I didn't want to hardcode picking only the first item so I present two options here:
- if the user wants to use a single item, they can use the built-in `index` function with an integer to select it. `{{ index .QueryParams.gopherAge 0 }}` returns `Blue`
- if they want all items, I provide the `stringsJoin` function to transform the slice into a delimited string. `{{ stringsJoin .QueryParams.gopherColor "," }}` returns `Blue,Purple`

Reply:
In the example I simply showcase both approaches

Reply:
Ah, yeah, right, my fault! 🙏🏻  All clear now 👌🏻 

Just to be clear, I read `index` connected with `.QueryParams` (as I wrote above), without realizing it's a function call, to get the data of the given index 😅